### PR TITLE
fix: video_player black screen

### PR DESCRIPTION
**Pull request by abdelaziz-mahdy**
i was testing video_player_media_kit to try out the seek without switching fully to media_kit i found out if the error stream throws any error even if its not critical (the sound will play but the video will be black and nothing will work as in ui wise no commands nor streams) but the sound works (so video got loaded correctly )

when i commented that part it works correctly, so i was thinking is it needed and if yes can we limit it to critical errors?

### CODE REVIEW

**Comment by alexmercerind** [comment]
I think we should instead optimize the logic here:

https://github.com/media-kit/media-kit/blob/bd6c4fd7f0aef553a02a439d1abd95a340fa8a52/media_kit/lib/src/player/native/player/real.dart#L2033-L2071

Feel free to remove the ones which are causing you an issue.

**Comment by abdelaziz-mahdy** [comment]
well thats what i got log `property not found` i think that will be ignored then (it needs more context i guess)

**Comment by abdelaziz-mahdy** [comment]
using log and error (i got a diffrent error :) )
(error stream)
flutter: log property not found 
(log stream)
flutter: log PlayerLog(prefix: lavf, level: error, text: Failed to create file cache.)

**Comment by abdelaziz-mahdy** [comment]
i ignored code -8 from throwing the error, since it was not the same location as you mentioned

**Comment by alexmercerind** [comment]
Not sure which invalid property we are trying to access...

**Comment by abdelaziz-mahdy** [comment]
> Not sure which invalid property we are trying to access...

will try to debug it more and let you know then

**Comment by abdelaziz-mahdy** [comment]
`flutter: _setProperty: name osc format 1 data is 5981231200 code is -8` this is the error location and call 
```
    print(
      "_setProperty: name $name format $format data is ${data.address} code is ${await completer.future}",
    );
    _error(await completer.future);
```

**Comment by alexmercerind** [comment]
Thanks a lot for taking a look. Maybe we can just comment out contents of this method:

https://github.com/media-kit/media-kit/blob/bd6c4fd7f0aef553a02a439d1abd95a340fa8a52/media_kit/lib/src/player/native/player/real.dart#L2474-L2479

**Comment by alexmercerind** [comment]
Callum added this recently in #815... We silently ignored these errors previously, I think it'll be good to keep the same.

**Comment by abdelaziz-mahdy** [comment]
what do you think about this change, it will give us the context of the errors in the logs, so we know if is a problem, and which command caused it

File: media_kit/lib/src/player/native/player/real.dart [hunk 1]
```diff
@@ -1355,7 +1355,8 @@ class NativePlayer extends PlatformPlayer {
       return;
     }
 
-    _error(event.ref.error);
+    _logError(event.ref.error,
+        'event:${event.ref.event_id} ${event.ref.data.cast<Uint8>()}');
```
Comment (alexmercerind) [inline media_kit/lib/src/player/native/player/real.dart:1359]: NIT: Trailing comma.

File: video_player_media_kit/lib/src/media_kit_video_player.dart [hunk 2]
```diff
@@ -334,12 +334,12 @@ class MediaKitVideoPlayer extends VideoPlayerPlatform {
...
-                code: '',
-                message: event,
-              ),
-            );
+            // streamController.addError(
+            //   PlatformException(
+            //     code: '',
+            //     message: event,
+            //   ),
+            // );
```
Comment (alexmercerind) [inline video_player_media_kit/lib/src/media_kit_video_player.dart]: It should be reverted.

File: media_kit/lib/src/player/native/player/real.dart [hunk 3]
```diff
@@ -2455,10 +2456,11 @@ class NativePlayer extends PlatformPlayer {
   }
 
   /// Adds an error to the [Player.stream.error].
-  void _error(int code) {
-    if (code < 0 && !errorController.isClosed) {
+  void _logError(int code, String? extraContext) {
```
Comment (alexmercerind) [inline media_kit/lib/src/player/native/player/real.dart:2459]: Can we rename `extraContext` to `text`?

```suggestion
  void _logError(int code, String? text) {
```

File: media_kit/lib/src/player/native/player/real.dart [hunk 4]
```diff
@@ -2455,10 +2456,11 @@ class NativePlayer extends PlatformPlayer {
...
+    if (code < 0 && !logController.isClosed) {
       final message = mpv.mpv_error_string(code).cast<Utf8>().toDartString();
-      errorController.add(message);
+      logController.add(PlayerLog(
+          prefix: "media_kit", level: 'error', text: 'error:$message $extraContext'));
```
Comment (alexmercerind) [inline media_kit/lib/src/player/native/player/real.dart:2463]: NIT: Trailing commas; single quote strings; adjust whitespaces.

```suggestion
          prefix: 'media_kit', level: 'error', text: 'error: $message text: $text'));
```

File: media_kit/lib/src/player/native/player/real.dart [hunk 5]
```diff
@@ -2482,12 +2484,14 @@ class NativePlayer extends PlatformPlayer {
       data,
     );
     calloc.free(namePtr);
+    String context = "_setProperty name $name format $format";
```
Comment (alexmercerind) [inline media_kit/lib/src/player/native/player/real.dart:2487]:
```suggestion
    final text = "_setProperty($name, $format)";
```

File: media_kit/lib/src/player/native/player/real.dart [hunk 6]
```diff
@@ -2482,12 +2484,14 @@ class NativePlayer extends PlatformPlayer {
...
     if (immediate < 0) {
       // Sending failed.
-      _error(immediate);
+      _logError(immediate, context);
```
Comment (alexmercerind) [inline media_kit/lib/src/player/native/player/real.dart:2490]:
```suggestion
      _logError(immediate, text);
```

File: media_kit/lib/src/player/native/player/real.dart [hunk 7]
```diff
@@ -2482,12 +2484,14 @@ class NativePlayer extends PlatformPlayer {
...
       return;
     }
-    _error(await completer.future);
+
+    _logError(await completer.future, context);
```
Comment (alexmercerind) [inline media_kit/lib/src/player/native/player/real.dart:2494]:
```suggestion
    _logError(await completer.future, text);
```

File: media_kit/lib/src/player/native/player/real.dart [hunk 8]
```diff
@@ -2547,12 +2551,14 @@ class NativePlayer extends PlatformPlayer {
     final immediate = mpv.mpv_command_async(ctx, requestNumber, arr.cast());
     calloc.free(arr);
     pointers.forEach(calloc.free);
+    String context = "_command ${args.join(" ")}";
     if (immediate < 0) {
       // Sending failed.
-      _error(immediate);
+      _logError(immediate, context);
```
Comment (alexmercerind) [inline media_kit/lib/src/player/native/player/real.dart:2557]:
```suggestion
      _logError(immediate, text);
```

File: media_kit/lib/src/player/native/player/real.dart [hunk 9]
```diff
@@ -2547,12 +2551,14 @@ class NativePlayer extends PlatformPlayer {
...
       return;
     }
-    _error(await completer.future);
+
+    _logError(await completer.future, context);
```
Comment (alexmercerind) [inline media_kit/lib/src/player/native/player/real.dart:2561]:
```suggestion
    _logError(await completer.future, text);
```

File: media_kit/lib/src/player/native/player/real.dart [hunk 10]
```diff
@@ -2547,12 +2551,14 @@ class NativePlayer extends PlatformPlayer {
     final immediate = mpv.mpv_command_async(ctx, requestNumber, arr.cast());
     calloc.free(arr);
     pointers.forEach(calloc.free);
+    String context = "_command ${args.join(" ")}";
```
Comment (alexmercerind) [inline media_kit/lib/src/player/native/player/real.dart:2554]:
```suggestion
    final text = "_command(${args.join(", ")})";
```

**Review by alexmercerind — CHANGES_REQUESTED** [review]
Thanks a lot. I hope it is working correctly for you now.

Minor requests from my end.

File: video_player_media_kit/lib/src/media_kit_video_player.dart (hunk 2 above)
Comment (alexmercerind) [reply video_player_media_kit/lib/src/media_kit_video_player.dart]: Resolved.

**Review by alexmercerind — APPROVED** [review]
LGTM!

**Comment by abdelaziz-mahdy** [comment]
> LGTM!
> 
> Feel free to merge when ready.

will test it again to be 100% sure then merge it

**Pull request merged**